File: C++/78 Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans(1, vector<int>());
        for (int num:nums) {
            int n = ans.size();
            for (int i = 0; i < n; i++) {
                vector<int> tmp = ans[i];
                tmp.push_back(num);
                ans.push_back(tmp);
            }
        }
        return ans;
    }
};

File: Python/78 Subsets.py
# Subsets
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        ans = [[]]
        for num in nums:
            ans += [ele + [num] for ele in ans]
        return ans